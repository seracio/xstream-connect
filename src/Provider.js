// @flow
import PropTypes from 'prop-types';
import React from 'react';

class Provider extends React.Component {

  constructor(props, context) {
    super(props, context);
    this.store = props.store;
  }

  getChildContext() {
    return { store: this.store };
  }

  render(){
    return React.Children.only(this.props.children);
  }
}

Provider.propTypes = {
  store: PropTypes.object.isRequired,
};

Provider.childContextTypes = {
  store: PropTypes.object.isRequired,
};

export default Provider;
