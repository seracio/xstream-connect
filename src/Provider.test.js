import React from 'react';
import renderer from 'react-test-renderer';
import Provider from './Provider';

test('Provider: only the child should be rendered', () => {
  const component = renderer.create(
    <Provider store={{test: 'test'}}>
      <div>hello world</div>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Provider: the child component should have a context with a key "store"', () => {
  class App extends React.Component {
    render(){
      return <div>{this.context.store.hello}</div>
    }
  }
  App.contextTypes = {
    store: React.PropTypes.object.isRequired
  };

  const component = renderer.create(
    <Provider store={{hello: 'hello world'}}>
      <App />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Provider: props store should be mandatory', () => {
  console.error = jest.fn();
  renderer.create(
    <Provider>
      <div>hello world</div>
    </Provider>
  );
  expect(console.error).toHaveBeenCalledTimes(2);
});


