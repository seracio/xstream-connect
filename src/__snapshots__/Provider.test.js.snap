exports[`test Provider: only the child should be rendered 1`] = `
<div>
  hello world
</div>
`;

exports[`test Provider: the child component should have a context with a key "store" 1`] = `
<div>
  hello world
</div>
`;
